runtime/util/annhelp.c

```diff
@@ -139,3 +143,206 @@ skipAnnotationElement(J9ROMConstantPoolItem const * constantPool, U_8 const *dat
+	J9ROMClass *romClass = NULL;
+	J9ROMConstantPoolItem *constantPool = NULL;
+	J9ROMFieldRef *romFieldRef = NULL;
+	J9ROMFieldShape *romFieldShape = NULL;
+
+	Assert_VMUtil_true(NULL != clazz);
+	Assert_VMUtil_true(NULL != annotationName);
+
+	romClass = clazz->romClass;
+	constantPool = J9_ROM_CP_FROM_ROM_CLASS(romClass);
+	romFieldRef = (J9ROMFieldRef *)&constantPool[cpIndex];
+	romFieldShape = getFieldShapeFromFieldRef(romClass, romFieldRef);
+
+	if (NULL != romFieldShape) {
+		U_32 *fieldAnnotationData = getFieldAnnotationsDataFromROMField(romFieldShape);
+
+		if (NULL != fieldAnnotationData) {
+			U_32 len = *fieldAnnotationData;
+			U_8 *data = (U_8 *)(fieldAnnotationData + 1);
+
+			annotationFound = findRuntimeVisibleAnnotation(data, len, annotationName, constantPool);
+		}
+	}
+
+	Trc_Util_annhelp_SearchForFieldAnnotation(J9UTF8_DATA(annotationName), cpIndex, clazz, romFieldShape, (UDATA) annotationFound);
+
+	return annotationFound;
+}
+
+/**
+ * Find the corresponding J9ROMFieldShape for a given J9ROMFieldRef.
+ *
+ * @param romClass The ROM Class the field belongs to.
+ * @param romFieldRef The Field Ref to find the matching Field Shape for.
+ * @return A pointer to the corresponding Field Shape or NULL if not found.
+ */
+static J9ROMFieldShape *
+getFieldShapeFromFieldRef(J9ROMClass *romClass, J9ROMFieldRef *romFieldRef)
+{
+	J9ROMFieldShape *romFieldShape = NULL;
+	J9ROMNameAndSignature *fieldRefNameAndSig = J9ROMFIELDREF_NAMEANDSIGNATURE(romFieldRef);
+	J9UTF8 *fieldRefName = J9ROMNAMEANDSIGNATURE_NAME(fieldRefNameAndSig);
+	J9UTF8 *fieldRefSignature = J9ROMNAMEANDSIGNATURE_SIGNATURE(fieldRefNameAndSig);
+	J9ROMFieldWalkState state;
+	J9ROMFieldShape *romFieldPtr = romFieldsStartDo(romClass, &state);
+	U_32 i;
+
+	for (i = 0; (i < romClass->romFieldCount) && (NULL != romFieldPtr); i++) {
+		J9UTF8 *fieldShapeName = J9ROMFIELDSHAPE_NAME(romFieldPtr);
+		J9UTF8 *fieldshapeSignature = J9ROMFIELDSHAPE_SIGNATURE(romFieldPtr);
+
+		if (J9UTF8_EQUALS(fieldRefName, fieldShapeName) && J9UTF8_EQUALS(fieldRefSignature, fieldshapeSignature)) {
+			romFieldShape = romFieldPtr;
+			break;
+		}
+
+		romFieldPtr = romFieldsNextDo(&state);
+	}
+
+	return romFieldShape;
+}
+
+/**
+ * Check if a method contains the specified Runtime Visible annotation.
+ *
+ * @param clazz The class the method belongs to.
+ * @param cpIndex The constant pool index of the method.
+ * @param annotationName The name of the annotation to check for.
+ * @return TRUE if the annotation is found, FALSE otherwise.
+ */
+BOOLEAN
+methodContainsRuntimeAnnotation(J9Class *clazz, UDATA cpIndex, J9UTF8 *annotationName)
+{
+	BOOLEAN annotationFound = FALSE;
+	J9ROMClass *romClass = NULL;
+	J9ROMConstantPoolItem *constantPool = NULL;
+	J9ROMMethodRef *romMethodRef = NULL;
+	J9ROMMethod *romMethod = NULL;
+
+	Assert_VMUtil_true(NULL != clazz);
+	Assert_VMUtil_true(NULL != annotationName);
+
+	romClass = clazz->romClass;
+	constantPool = J9_ROM_CP_FROM_ROM_CLASS(clazz->romClass);
```

Comment:
Babneet called this out but it didn't get fixed. `romClass` can be used rather than `clazz->romClass`, since `romClass` is set on the previous line.